### ISSUE TITLE
Successful crawls are no longer outputting new CSV files

### ISSUE THREAD

**Issue by novellac** [City-Bureau/city-scrapers#432]
[432-Traceback-MissingCsvOutput.txt](https://github.com/City-Bureau/city-scrapers/files/2178499/432-Traceback-MissingCsvOutput.txt)

Issue: Successful crawls are no longer outputting new CSV files

Expected Result: When the command `scrapy crawl <spider name>` is run, a CSV file should be output to the folder `/city-scrapers/city-scrapers/local_outputs/`

Hints: May be a recently deprecated library. As per @bonfirefan and @jim, we may have updated our libraries at some point and need to add in the `pydispatch` library into our dependencies.

**Comment by bonfirefan** [comment]
So, I tested adding in pydispatch and it doesn't seem to be the issue here.

**Comment by bonfirefan** [comment]
@pjsier could this be related to the [s3 pipeline](https://github.com/City-Bureau/city-scrapers/blob/master/city_scrapers/pipelines/s3_item.py)?

**Comment by pjsier** [comment]
@bonfirefan when I uncomment the CSV pipeline in `city_scrapers/settings/base.py` it works for me, otherwise I can't reproduce this. We might want to leave this on by default if it doesn't hurt anything for local development

**Comment by bonfirefan** [comment]
@pjsier no wonder, that explains the local output portion. I think these are possibly two separate errors - it would be good to uncomment out the local exporter pipeline unless it was causing other errors?

Separate issue below. Here's what I'm seeing even after successfully outputting to csv after uncommenting the csv pipeline is output that matches @novellac's error output:

```
2018-07-09 20:27:07 [botocore.session] DEBUG: Loading variable profile from defaults.
2018-07-09 20:27:07 [botocore.session] DEBUG: Loading variable config_file from defaults.
2018-07-09 20:27:07 [botocore.session] DEBUG: Loading variable credentials_file from defaults.
2018-07-09 20:27:07 [botocore.session] DEBUG: Loading variable data_path from defaults.
2018-07-09 20:27:07 [botocore.session] DEBUG: Loading variable profile from defaults.
2018-07-09 20:27:07 [botocore.session] DEBUG: Loading variable region from defaults.
2018-07-09 20:27:07 [botocore.session] DEBUG: Loading variable profile from defaults.
2018-07-09 20:27:07 [botocore.session] DEBUG: Loading variable ca_bundle from defaults.
2018-07-09 20:27:07 [botocore.session] DEBUG: Loading variable profile from defaults.
2018-07-09 20:27:07 [botocore.session] DEBUG: Loading variable api_versions from defaults.
2018-07-09 20:27:07 [botocore.session] DEBUG: Loading variable profile from defaults.
2018-07-09 20:27:07 [botocore.session] DEBUG: Loading variable credentials_file from defaults.
2018-07-09 20:27:07 [botocore.session] DEBUG: Loading variable config_file from defaults.
2018-07-09 20:27:07 [botocore.session] DEBUG: Loading variable profile from defaults.
2018-07-09 20:27:07 [botocore.session] DEBUG: Loading variable metadata_service_timeout from defaults.
2018-07-09 20:27:07 [botocore.session] DEBUG: Loading variable profile from defaults.
2018-07-09 20:27:07 [botocore.session] DEBUG: Loading variable metadata_service_num_attempts from defaults.
2018-07-09 20:27:07 [botocore.session] DEBUG: Loading variable profile from defaults.
2018-07-09 20:27:07 [botocore.credentials] DEBUG: Looking for credentials via: env
2018-07-09 20:27:07 [botocore.credentials] DEBUG: Looking for credentials via: assume-role
2018-07-09 20:27:07 [botocore.credentials] DEBUG: Looking for credentials via: shared-credentials-file
2018-07-09 20:27:07 [botocore.credentials] DEBUG: Looking for credentials via: config-file
2018-07-09 20:27:07 [botocore.credentials] DEBUG: Looking for credentials via: ec2-credentials-file
2018-07-09 20:27:07 [botocore.credentials] DEBUG: Looking for credentials via: boto-config
2018-07-09 20:27:07 [botocore.credentials] DEBUG: Looking for credentials via: container-role
2018-07-09 20:27:07 [botocore.credentials] DEBUG: Looking for credentials via: iam-role
2018-07-09 20:27:07 [botocore.vendored.requests.packages.urllib3.connectionpool] INFO: Starting new HTTP connection (1): 169.254.169.254
2018-07-09 20:27:08 [botocore.utils] DEBUG: Caught exception while trying to retrieve credentials: HTTPConnectionPool(host='169.254.169.254', port=80): Max retries exceeded with url: /latest/meta-data/iam/security-credentials/ (Caused by ConnectTimeoutError(<botocore.awsrequest.AWSHTTPConnection object at 0x7fe9cac5ac18>, 'Connection to 169.254.169.254 timed out. (connect timeout=1)'))
Traceback (most recent call last):
  File "/home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/vendored/requests/packages/urllib3/connection.py", line 134, in _new_conn
    (self.host, self.port), self.timeout, **extra_kw)
  File "/home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/vendored/requests/packages/urllib3/util/connection.py", line 88, in create_connection
    raise err
  File "/home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/vendored/requests/packages/urllib3/util/connection.py", line 78, in create_connection
    sock.connect(sa)
socket.timeout: timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/vendored/requests/packages/urllib3/connectionpool.py", line 544, in urlopen
    body=body, headers=headers)
  File "/home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/vendored/requests/packages/urllib3/connectionpool.py", line 349, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/lib/python3.5/http/client.py", line 1106, in request
    self._send_request(method, url, body, headers)
  File "/home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/awsrequest.py", line 130, in _send_request
    self, method, url, body, headers, *args, **kwargs)
  File "/usr/lib/python3.5/http/client.py", line 1151, in _send_request
    self.endheaders(body)
  File "/usr/lib/python3.5/http/client.py", line 1102, in endheaders
    self._send_output(message_body)
  File "/home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/awsrequest.py", line 157, in _send_output
    self.send(msg)
  File "/home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/awsrequest.py", line 242, in send
    return HTTPConnection.send(self, str)
  File "/usr/lib/python3.5/http/client.py", line 877, in send
    self.connect()
  File "/home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/vendored/requests/packages/urllib3/connection.py", line 155, in connect
    conn = self._new_conn()
  File "/home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/vendored/requests/packages/urllib3/connection.py", line 139, in _new_conn
    (self.host, self.timeout))
botocore.vendored.requests.packages.urllib3.exceptions.ConnectTimeoutError: (<botocore.awsrequest.AWSHTTPConnection object at 0x7fe9cac5ac18>, 'Connection to 169.254.169.254 timed out. (connect timeout=1)')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/vendored/requests/adapters.py", line 370, in send
    timeout=timeout
  File "/home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/vendored/requests/packages/urllib3/connectionpool.py", line 597, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/vendored/requests/packages/urllib3/util/retry.py", line 271, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
botocore.vendored.requests.packages.urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='169.254.169.254', port=80): Max retries exceeded with url: /latest/meta-data/iam/security-credentials/ (Caused by ConnectTimeoutError(<botocore.awsrequest.AWSHTTPConnection object at 0x7fe9cac5ac18>, 'Connection to 169.254.169.254 timed out. (connect timeout=1)'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/utils.py", line 174, in _get_request
    response = requests.get(url, timeout=timeout)
  File "/home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/vendored/requests/api.py", line 69, in get
    return request('get', url, params=params, **kwargs)
  File "/home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/vendored/requests/api.py", line 50, in request
    response = session.request(method=method, url=url, **kwargs)
  File "/home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/vendored/requests/sessions.py", line 465, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/vendored/requests/sessions.py", line 573, in send
    r = adapter.send(request, **kwargs)
  File "/home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/vendored/requests/adapters.py", line 419, in send
    raise ConnectTimeout(e, request=request)
botocore.vendored.requests.exceptions.ConnectTimeout: HTTPConnectionPool(host='169.254.169.254', port=80): Max retries exceeded with url: /latest/meta-data/iam/security-credentials/ (Caused by ConnectTimeoutError(<botocore.awsrequest.AWSHTTPConnection object at 0x7fe9cac5ac18>, 'Connection to 169.254.169.254 timed out. (connect timeout=1)'))
2018-07-09 20:27:08 [botocore.utils] DEBUG: Max number of attempts exceeded (1) when attempting to retrieve data from metadata service.
2018-07-09 20:27:08 [botocore.loaders] DEBUG: Loading JSON file: /home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/data/endpoints.json
2018-07-09 20:27:08 [botocore.session] DEBUG: Loading variable profile from defaults.
2018-07-09 20:27:08 [botocore.loaders] DEBUG: Loading JSON file: /home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/data/s3/2006-03-01/service-2.json
2018-07-09 20:27:08 [botocore.hooks] DEBUG: Event creating-client-class.s3: calling handler <function add_generate_presigned_post at 0x7fe9cacbc8c8>
2018-07-09 20:27:08 [botocore.hooks] DEBUG: Event creating-client-class.s3: calling handler <function lazy_call.<locals>._handler at 0x7fe9cd09e2f0>
2018-07-09 20:27:08 [botocore.hooks] DEBUG: Event creating-client-class.s3: calling handler <function add_generate_presigned_url at 0x7fe9cacbc6a8>
2018-07-09 20:27:08 [botocore.args] DEBUG: The s3 config key is not a dictionary type, ignoring its value of: None
2018-07-09 20:27:08 [botocore.endpoint] DEBUG: Setting s3 timeout as (60, 60)
2018-07-09 20:27:08 [botocore.loaders] DEBUG: Loading JSON file: /home/bfan/.local/share/virtualenvs/documenters-aggregator/lib/python3.5/site-packages/botocore/data/_retry.json
2018-07-09 20:27:08 [botocore.client] DEBUG: Registering retry handlers for service: s3
2018-07-09 20:27:08 [botocore.client] DEBUG: Defaulting to S3 virtual host style addressing with path style addressing fallback.
2018-07-09 20:27:08 [scrapy.middleware] INFO: Enabled item pipelines:
['city_scrapers.pipelines.CityScrapersItemPipeline',
 'city_scrapers.pipelines.CsvPipeline']
2018-07-09 20:27:08 [scrapy.core.engine] INFO: Spider opened
2018-07-09 20:27:08 [scrapy.extensions.logstats] INFO: Crawled 0 pages (at 0 pages/min), scraped 0 items (at 0 items/min)
2018-07-09 20:27:08 [scrapy.extensions.telnet] DEBUG: Telnet console listening on 127.0.0.1:6023
2018-07-09 20:27:09 [scrapy.core.engine] DEBUG: Crawled (404) <GET https://www.cityofchicago.org/robots.txt> (referer: None)
```
[bf_log_error.txt](https://github.com/City-Bureau/city-scrapers/files/2178526/bf_log_error.txt)

**Comment by pjsier** [comment]
That's odd... can anyone else reproduce this? I'm wondering if it could have to do with an old version of boto or not having AWS credentials set up locally

**Comment by r-wei** [comment]
hey @bonfirefan , what branch are you running to create that error? bonfirefan:pydispatch-export?

**Comment by bonfirefan** [comment]
@r-wei this error appears on Master, but I tried to see if adding libraries or uncommenting out the csv exporter were the issue. Uncommenting out the csv exporter solved the issue of csv exports (unsurprisingly) and I think @novellac and I were just seeing additional errors that probably have to do with the s3 pipeline - that being said, these errors don't seem to affect the crawlers at all. If you're able to replicate them on Master with the csv exporter commented out, then it's AWS related

**Comment by r-wei** [comment]
yes, I also get the connect timeout error at the beginning, but the spider still scrapes and writes a csv

**Comment by pjsier** [comment]
Closed by #434
